src/event_logger.cpp

```diff
@@ -60,17 +60,90 @@ void EventLogger::createAsyncPelWithInventoryCallout(
+        {
+            logging::logMessage(
+                "PEL error type not found in the error message map.");
+            return;
+        }
+
+        const std::string& l_message = m_errorMsgMap.at(i_errorType);
+
+        const std::string& l_severity =
+            (m_severityMap.find(i_severity) != m_severityMap.end()
+                 ? m_severityMap.at(i_severity)
+                 : m_severityMap.at(types::SeverityType::Error));
+
+        std::string l_description("Description field missing in user data");
+
+        if (i_userData1 && ((*i_userData1).first.compare("DESCRIPTION") ==
+                            constants::STR_CMP_SUCCESS))
+        {
+            l_description = (*i_userData1).second;
+        }
+
+        sd_bus* l_sdBus = nullptr;
+        sd_bus_default(&l_sdBus);
+
+        const types::InventoryCalloutData& l_invCallout = i_callouts[0];
+        const types::CalloutPriority& l_priorityEnum = get<1>(l_invCallout);
+
+        const std::string& l_priority =
+            (m_priorityMap.find(l_priorityEnum) != m_priorityMap.end()
+                 ? m_priorityMap.at(l_priorityEnum)
+                 : m_priorityMap.at(types::CalloutPriority::Low));
+
+        auto l_rc = -1;
+        if (!i_userData2)
+        {
+            l_rc = sd_bus_call_method_async(
+                l_sdBus, NULL, constants::eventLoggingServiceName,
+                constants::eventLoggingObjectPath,
+                constants::eventLoggingInterface, "Create", NULL, NULL,
+                "ssa{ss}", l_message.c_str(), l_severity.c_str(), 6, "FileName",
+                i_fileName.c_str(), "FunctionName", i_funcName.c_str(),
+                "InternalRc", i_internalRc.c_str(), "DESCRIPTION",
+                l_description.c_str(), "CALLOUT_INVENTORY_PATH",
+                get<0>(l_invCallout).c_str(), "CALLOUT_PRIORITY",
+                l_priority.c_str());
+        }
+        else
+        {
+            l_rc = sd_bus_call_method_async(
+                l_sdBus, NULL, constants::eventLoggingServiceName,
+                constants::eventLoggingObjectPath,
+                constants::eventLoggingInterface, "Create", NULL, NULL,
+                "ssa{ss}", l_message.c_str(), l_severity.c_str(), 7, "FileName",
+                i_fileName.c_str(), "FunctionName", i_funcName.c_str(),
+                "InternalRc", i_internalRc.c_str(), "DESCRIPTION",
+                l_description.c_str(), (*i_userData2).first.c_str(),
+                (*i_userData2).second.c_str(), "CALLOUT_INVENTORY_PATH",
+                get<0>(l_invCallout).c_str(), "CALLOUT_PRIORITY",
+                l_priority.c_str());
+        }
+
+        if (l_rc < 0)
+        {
+            logging::logMessage(
+                "Error calling sd_bus_call_method_async, Message = " +
+                std::string(strerror(-l_rc)));
+        }
+    }
+    catch (const sdbusplus::exception::SdBusError& l_ex)
+    {
+        logging::logMessage(l_ex.what());
```

Comment:
just remind me please on why this try catch , what functions will be giving exceptions in this case.

Reply:
if any exception occurs in this API, instead throwing back to caller its handled here by printing the error and returning caller quietly.
Updated the exception.